### DISCUSSION TITLE
Get id of a q-tree item under mouse

### DISCUSSION THREAD

**Discussion by denprog** [quasarframework/quasar#16588] [General - Components / Directives / etc]
I want to have a context menu above a q-tree. I created both:
`
<template>

    <div class="q-sm">
        <q-input class="q-pa-sm" dense ref="documentsFilterRef" v-model="documentsFilter">
            <template v-slot:append>
                <q-icon v-if="documentsFilter !== ''" name="clear" class="cursor-pointer" @click="resetDocumentsFilter" />
            </template>
        </q-input>
        <q-tree :nodes="documents" dense v-model:selected="selectedDocument" ref="documentsRef" node-key="id" label-key="label" 
            :filter="documentsFilter" @update:selected="onDocumentSelected" default-expand-all>
        </q-tree>
        <q-menu touch-position square context-menu auto-close>
            <q-list dense style="min-width: 100px">
                <q-item id="rename-menu" clickable @click='onRename();'>
                    <q-item-section>Rename</q-item-section>
                </q-item>
                <q-item id="delete-menu" clickable @click='onDelete();'>
                    <q-item-section>Delete</q-item-section>
                </q-item>
            </q-list>
        </q-menu>
    </div>

</template>
`
Now in the handler onDelete or onRename I'd like to get id of the tree item below the mouse. How can I do that? In this time the item is not selected.

**Comment by denprog** [comment]
Is there any way to make a context menu for tree or list like on the desktop?

**Reply by mickey-bimlive** [reply]
Suggest to insert a v-slot for the header of the q-tree nodes, right after the first q-tree tag, like:

```
<q-tree ...>
 <template v-slot:default-header="{ node }">
    <!-- Display id of node: -->
    <div class="text-caption"> id of node: {{ node.id }} </div>
    <!-- Context menu for node: -->
    <q-menu  touch-position  context-menu>
      <q-list>
         <q-item clickable v-close-popup @click="doSomething(node.id)">
            <q-item-section>Do something</q-item-section>
         </q-item>
         <!-- Add other q-items here, for other menu items, like delete, rename node, etc. -->
      </q-list>
   </q-menu>
</template>
</q-tree>
```
